core/util.py

```diff
@@ -58,13 +59,36 @@ def group_by_first_element(tuples: list[tuple]) -> dict[str, list[tuple | Any]]:
     return nested
 
 
-def get_project_number(project_id):
-    """Extracts the project number from a project ID.
+def get_project_number_by_cell(row_index: int, sheet: str):
+    """Extracts the project number from the row index and table name of a row by mapping it's on page position
+
+    :param row_index: row number on Excel sheet that row has come from
+    :param sheet: name of the database table the row has come from
+    :return: project number
+    """
+    match sheet:
+        case "Funding" | "Funding Comments":
+            project_number = math.ceil((row_index - 32) / 28)
+        case "Output_Data":
+            project_number = math.ceil((row_index - 17) / 38)
+        case "RiskRegister":
+            project_number = math.ceil((row_index - 19) / 8) if row_index < 44 else math.ceil((row_index - 20) / 8)
+        case _:
+            project_number = None
```

Comment:
Do we want this to fail silently? The other project number function will raise an exception. I think it's best to keep the behaviour consistent.